notebook/notebookapp.py

```diff
@@ -1432,14 +1432,16 @@ def display_url(self):
                 url += '/'
         else:
             if self.ip in ('', '0.0.0.0'):
-                ip = "(%s or 127.0.0.1)" % socket.gethostname()
+                ip = "%s" % socket.gethostname()
             else:
                 ip = self.ip
             url = self._url(ip)
         if self.token:
             # Don't log full token if it came from config
             token = self.token if self._token_generated else '...'
-            url = url_concat(url, {'token': token})
+            url = (url_concat(url, {'token': token})
+                  + '\n or '
+                  + url_concat(self._url('127.0.0.1'), {'token': token}))
```

Comment:
We've previously been trying to show only the real IP if there's one set, and make the 'or 127.0.0.1' a special case when you're listening on all IPs. This would show the 127.0.0.1 option all the time, even when you're not listening on that IP, and even when it's the same as the first address. But maybe that's better than trying to be clever about it?

Reply:
Hi @takluyver, thanks for the update.
In my experience it would be an unusual system not to be listening on 127.0.0.1, the loopback address. Is that an existing use case for jupyter notebooks?

Reply:
It's definitely possible to listen only on an external-facing interface, though I don't know how many people do it. But even when you do that, it might be e.g. the external network interface of a container, which is mapped to the same port on localhost outside the container. There's no way in general to know what address the server is actually accessible at.

So I'm inclined to merge this, and we can always reconsider it again later.